internal/state/graph/graph.go

```diff
@@ -50,7 +51,7 @@ func BuildGraph(
 
 	processedGws := processGateways(state.Gateways, gcName)
 
-	gw := buildGateway(processedGws.Winner, secretMemoryMgr, gc)
+	gw := buildGateway(processedGws.Winner, secretMemoryMgr, gc, state.ReferenceGrants)
```

Comment:
would it make sense to add ReferenceGrant to the test in `graph_test.go`?

Reply:
Do you think we need both a positive (ref grant exists) and negative (ref grant doesn't exist) test case for the graph? 

The negative test seems redundant since we test that in `gateway_test.go` and `reference_grant_test.go`.

Reply:
> The negative test seems redundant since we test that in gateway_test.go and reference_grant_test.go.

👍